File: src/main/java/org/pl/parser/ast/VarDeclarationNode.java
package org.pl.parser.ast;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;


/**
 * Node that enables variable declaration.
 */
public class VarDeclarationNode implements INode {
    public final List<VarDeclaration> declarations;
    public final List<Modifier> modifiers = new ArrayList<>(4);

    public VarDeclarationNode(List<VarDeclaration> declarations, Modifier... modifiers) {
        this.declarations = declarations;
        this.modifiers.addAll(Arrays.asList(modifiers));
    }

    @Override
    public String toString() {
        return "VarDeclarationNode{" +
                "declarations=" + declarations +
                ", modifiers=" + modifiers +
                '}';
    }
}

File: src/main/java/org/pl/parser/ast/VarDeclarationNode.kt
package org.pl.parser.ast

/**
 * Node that enables variable declaration for multiple variables at a time.
 *
 * Example:
 * `var a: bool, b = 10, c: number;`
 */
class VarDeclarationNode(val declarations: List<VarDeclaration>, vararg modifiers: Modifier) : INode {

    val modifiers = arrayListOf<Modifier>()

    init {
        this.modifiers.addAll(listOf(*modifiers))
    }

    override fun toString() = "VarDeclarationNode{declarations=$declarations, modifiers=$modifiers}"
}